packages/react/README.md

```diff
@@ -101,3 +119,593 @@ function WidgetDeleter() {
+  // fetch only the widget id and name fields
+  const [{ data, fetching, error }, _refetch] = useFindOne(api.blogPost, props.id, {
+    select: {
+      id: true,
+      title: true,
+      body: true,
+      // fetch the post's `comments` HasMany relationship field on the post
+      comments: {
+        edges: {
+          node: {
+            id: true,
+            body: true,
+            // fetch the author's BelongsTo User relationship field also
+            author: {
+              email: true,
+            },
+          },
+        },
+      },
+    },
+  });
+
+  if (!data) return null;
+  return (
+    <>
+      <h2>{data.title}</h2>
+      <ul>
+        {data.comments.edges.map((edge) => (
+          <li>
+            {edge.node.author.email} says {edge.node.body}
+          </li>
+        ))}
+      </ul>
+    </>
+  );
+};
+```
+
+**Note**: The shape of the options you pass in the `select` option matches exactly the shape of the GraphQL API for your application. Gadget applications use Relay-style GraphQL pagination, which means lists of records are accessed using the `relatedField: { edges: { node: true } }` style. BelongsTo and HasOne field types are accessed without any intermediate fields.
+
+For TypeScript users, the `select` option is fully typesafe, allowing you to typecheck which fields you're fetching from the backend as well as ensure that the fields you render in your components are actually selected.
+
+### Errors from the returned `error` object
+
+Running queries or mutations can produce a few different kinds of errors your client side should handle:
+
+- network errors where the browser is unable to connect to the server at all
+- validation errors where the client sent information to the server successfully, but the server deemed it invalid and rejected it
+- server side errors where the client sent information to the server but the server failed to process it due to a bug or transient issue.
+
+Each of these error cases is broken out on the `error` object returned by `useAction` (and any of the other hooks). The `error` object is an `ErrorWrapper` object, which has a number of properties for figuring out exactly what went wrong:
+
+- `error.message: string` - A top level error message which is always present
+- `error.networkError: Error | undefined` - An error thrown by the browser when trying to communicate with the server
+- `error.executionErrors: (GraphQLError | GadgetError)[] | undefined` - Any errors thrown by the GraphQL API, like missing parameters or invalid selections, and any errors thrown by the server concerning invalid data or backend processing errors.
+- `error.validationErrors: { apiIdentifier: string, message: string }[] | undefined` - Any validation errors returned by the server. A shortcut to accessing the `.validationErrors` property of the first `InvalidRecordError` in the `.executionErrors` of the outer `ErrorWrapper` object. Useful for building form validations.
+
+#### Default selections
```

Comment:
`###` instead of `####`? This doesn't look like a subsection of "Errors from the returned `error` object"